app/(kiosk display)/kiosks/[slug]/DepartureItemTime.tsx

```diff
@@ -0,0 +1,36 @@
+import clsx from 'clsx';
+import styles from './DepartureItem.module.css';
+import { DepartureTime } from '../../../../types/kioskDisplayTypes/GroupedRoute';
+import RealTimeIcon from './RealTimeIcon';
+import { useRecoilValue } from 'recoil';
+import { darkModeState } from '../../../../state/kioskState';
+
+interface DepartureItemTimeProps {
+	time: DepartureTime;
+	index: number;
+}
+
+export default function DepartureItemTime({ time, index }: DepartureItemTimeProps) {
+	const timeAsInt = parseInt(time.time.split(' ')[0]);
+	const isDarkMode = useRecoilValue(darkModeState);
+	const isSoon = time.time == 'DUE' || (!isNaN(timeAsInt) && timeAsInt < 5);
+
+	const timeClasses = clsx('departure-time', styles.time, { [styles.soonDark]: isDarkMode && isSoon }, { [styles.soonLight]: !isDarkMode && isSoon });
+	const departureTimeClasses = clsx('departureTime', styles.departureTime);
+	const timeSubtitleClasses = clsx('timeSubtitle', styles.timeSubtitle);
+	const realtimeIconClasses = clsx('realtimeIcon', styles.realtimeIcon);
+
+	return (
+		<div className={clsx(departureTimeClasses, { [styles.hopper]: time.isHopper })} key={index}>
```

Comment:
Index should not be used as a key.

```suggestion
		<div className={clsx(departureTimeClasses, { [styles.hopper]: isHopper })} key={`${time}-${isHopper}-${modifier}`}>
```